bundler/lib/dependabot/bundler/file_parser.rb

```diff
@@ -327,6 +335,51 @@ def imported_ruby_files
           .reject { |f| f.name == "gems.rb" }
       end
 
+      sig { returns(String) }
+      def bundler_raw_version
+        return bundler_raw_version if defined?(@bundler_raw_version)
+
+        package_manager = PackageManager.new(bundler_version)
+
+        # If selected version is unsupported, we are going to throw unsupported error
+        # So we shoudn't try to get the raw version
+        return bundler_version if package_manager.unsupported?
+
+        # read raw version directly from the ecosystem environment
+        bundler_raw_version = SharedHelpers.in_a_temporary_repo_directory(
```

Comment:
**Review Tip:** When retrieving the full raw version of Bundler, note that if an unsupported Bundler version (e.g., `"1"`) is specified, retrieving the raw version won’t be possible. In such cases, an "unsupported version" is getting raised error so we won't need to fetch the raw version.